### PR TITLE
Nieto/asset profile fe

### PR DESCRIPTION
**Pull request by JMNietoatBusyBee**
"Asset Profile Module"

### CODE REVIEW

File: src/app/core/services/portal.service.ts [hunk 1]
```diff
@@ -249,7 +249,19 @@ export class PortalService {
...
+    return this._portalApi.addAssetProfile(assetProfile);
+}
+
+  updateAssetProfile(id: number , data:any): Observable<any> {
+    return this._portalApi.updateAssetProfile(id,data);
+}
+
+getAssetProfile(id: number): Observable<IAssetProfile> {
+  return this._portalApi.getAssetProfile(id);
+}
```
Comment (ferlyncaps) [inline src/app/core/services/portal.service.ts:266]: just fix the alignment here based on the one above it :)

File: src/app/pages/apps/asset-profile/asset-profile.component.ts [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+      isActive: assetData.isActive === 'Y' ? true : false
+    })
+    this.createdBy = assetData.createdByName
+    this.createdAt = assetData.createdAt
+    this.updatedBy = assetData.updatedByName || 'NA';
+    this.updatedAt = assetData.updatedAt
+    this.statusDate = assetData.statusDate
+   }
+
+   private _getRoleAuthorization(roles: any[]): any {
```
Comment (ferlyncaps) [inline src/app/pages/apps/asset-profile/asset-profile.component.ts:268]: adjust alignment

**Review by ferlyncaps — CHANGES_REQUESTED** [review]

File: src/app/core/services/portal.service.ts (hunk 1 above)
Comment (JMNietoatBusyBee) [reply src/app/core/services/portal.service.ts:266]: applied.

File: src/app/pages/apps/asset-profile/asset-profile.component.ts [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+  name: ['', Validators.required], 
+  rcaglId: ['', Validators.required],
+  assetCategoryId: [''],
+  rcaSLId: ['',Validators.required],
+  assetClassId: ['', [Validators.required, Validators.pattern('^[0-9]*$')]],
+  costcenter: ['',Validators.required],
+  description: [''],
+  economiclife: ['',Validators.required],
+  residuallife: ['',Validators.required],
+  isActive: ['Y', []]
```
[the remark this answers is not in the document]
Comment (JMNietoatBusyBee) [reply src/app/pages/apps/asset-profile/asset-profile.component.ts:75]: Applied.

File: src/app/pages/apps/asset-profile/asset-profile.component.ts (hunk 2 above)
Comment (JMNietoatBusyBee) [reply src/app/pages/apps/asset-profile/asset-profile.component.ts:268]: Applied.

**Review by ferlyncaps — APPROVED** [review]

**Pull request merged**